kedro/extras/datasets/pandas/gbq_dataset.py

```diff
@@ -161,3 +169,145 @@ def _validate_location(self):
                 "to be the same for save and load args. "
                 "Details: https://cloud.google.com/bigquery/docs/locations"
             )
+
+
+class GBQQueryDataSet(AbstractDataSet):
+    """``GBQQueryDataSet`` loads data from a provided SQL query from Google
+    BigQuery. It uses ``pandas.read_gbq`` which itself uses ``pandas-gbq``
+    internally to read from BigQuery table. Therefore it supports all allowed
+    pandas options on ``read_gbq``.
+
+    Example adding a catalog entry with
+    `YAML API <https://kedro.readthedocs.io/en/stable/05_data/\
+        01_data_catalog.html#using-the-data-catalog-with-the-yaml-api>`_:
+
+    .. code-block:: yaml
+
+        >>> vehicles:
+        >>>   type: pandas.GBQQueryDataSet
+        >>>   sql: "select shuttle, shuttle_id from spaceflights.shuttles;"
+        >>>   project: my-project
+        >>>   credentials: gbq-creds
+        >>>   load_args:
+        >>>     reauth: True
+        >>>   save_args:
+        >>>     chunk_size: 100
```

Comment:
```suggestion
```

The dataset doesn't actually support saving or `save_args` so this example is a bit misleading.